lib/model/model_test.go

```diff
@@ -947,11 +947,13 @@ func TestIgnores(t *testing.T) {
 
 	db := db.OpenMemory()
 	m := NewModel(defaultConfig, protocol.LocalDeviceID, "device", "syncthing", "dev", db, nil)
-	m.AddFolder(defaultFolderConfig)
-	m.StartFolder("default")
 	m.ServeBackground()
+	m.cfg.SetFolder(defaultFolderConfig)
 	defer m.Stop()
 
+	// wait for goroutine wrapper.notifyListener to finish
+	time.Sleep(time.Duration(5) * time.Millisecond)
```

Comment:
OK! Now that we know what we're waiting for:

```
type configWaiter struct {
    done chan struct{}
}

func (configWaiter) VerifyConfiguration(from, to Configuration) error { return nil }
func (w *configWaiter) CommitConfiguration(from, to Configuration) (handled bool) {
    close(w.done)
    return true
}
func (configWaiter) String() string { return "configWaiter" }
```

and then instead of the sleep

```go
...
defer m.Stop()

configDone := make(chan struct{})
m.cfg.Subscribe(&configWater{configDone})
m.cfg.SetFolder(defaultFolderConfig)
<-configDone
```

(type directly into browser, probably buggy, but principle should be sound)